Index: DOTween/Tween.cs
<+>UTF-8
===================================================================
diff --git a/DOTween/Tween.cs b/DOTween/Tween.cs
--- a/DOTween/Tween.cs
+++ b/DOTween/Tween.cs
@@ -227,6 +227,16 @@
         }


+
+
+        public bool HasTargetId { get; set; }
+
+
+
+
+        public int TargetId { get; set; }
+
+
         public float timeScale;


Index: DOTween/Core/TweenManager.cs
<+>UTF-8
===================================================================
diff --git a/DOTween/Core/TweenManager.cs b/DOTween/Core/TweenManager.cs
--- a/DOTween/Core/TweenManager.cs
+++ b/DOTween/Core/TweenManager.cs
@@ -483,7 +483,14 @@
                         flag2 = true;
                         break;
                     case FilterType.TargetOrId:
-                        flag2 = id.Equals(tween.id) || id.Equals(tween.target);
+                        if (operationType == OperationType.Despawn && tween.HasTargetId)
+                        {
+                            flag2 = id.Equals(tween.TargetId);
+                        }
+                        else
+                        {
+                            flag2 = id.Equals(tween.id) || id.Equals(tween.target);
+                        }
                         break;
                     case FilterType.TargetAndId:
                         flag2 = id.Equals(tween.id) && optionalObj != null && optionalObj.Equals(tween.target);